### PR TITLE
doc: Release process

### PR DESCRIPTION
**Pull request by stevenhorsman**
- Rework our release process documentation based on the change of process in CoCo

### CODE REVIEW

File: docs/Release-Process.md [hunk 1]
```diff
@@ -5,44 +5,69 @@ Containers release
...
-- The attestation-agent that is built into the peer pod vm image, by updating the `git.guest-components.reference` value in [versions.yaml](../src/cloud-api-adaptor/versions.yaml)
+      - The `kata-containers/src/runtime` go module that we include in the main `cloud-api-adaptor` [`go.mod`](../src/cloud-api-adaptor/go.mod) and the `csi-wrapper` [`go.mod`](../src/csi-wrapper/go.mod). This can be done by running
+        ```
+        go get github.com/kata-containers/kata-containers/src/runtime@<latest release e.g. 3.6.0>
+        go mod tidy
+        ```
+        in the [cloud-api-adaptor](../src/cloud-api-adaptor/) directory and [csi-wrapper](../src/csi-wrapper/) directory.
+1. Trustee releases?
+    - We should bump the `kbs` version in [versions.yaml](../src/cloud-api-adaptor/versions.yaml)
+     to match the released and tested version of trustee, if applicable
```
Comment (fitzthum) [inline docs/Release-Process.md:34]: This should probably be coupled with the releases of guest-components. i think we will almost always tag those together.

File: docs/Release-Process.md [hunk 2]
```diff
@@ -5,44 +5,69 @@ Containers release
...
+1. cloud-api-adaptor releases with the following phases detailed below:
+    - Pre-release testing
+    - Cutting the release
+    - Post release tasks
+1. The CoCo operator updates to use references to the other component releases and then releases itself
```
Comment (fitzthum) [inline docs/Release-Process.md:39]: If the operator release needs to happen after the CAA stuff, we should add something to the release checklist.

Is the step required in the operator for peer pods just bumping the payloads in the remote CRD yaml?

Comment (stevenhorsman) [reply docs/Release-Process.md:39]: So for this release bumping the operator first is fine as we had to roll back the changes to get the peerpod controllers managed by the operator, but if/when we resolve the challenges there we would need to do peer pod before the operator, so we can update the peer pods references like Pradipta did in https://github.com/confidential-containers/operator/pull/362

File: docs/Release-Process.md (hunk 1 above)
Comment (stevenhorsman) [reply docs/Release-Process.md]: Thanks for the info. I've updated the doc in a separate commit to make it easier to re-check, but will fixup before this gets merged.

File: docs/Release-Process.md [hunk 3]
```diff
@@ -5,44 +5,66 @@ Containers release
...
+In the new Confidential Containers
+[release process](https://github.com/confidential-containers/community/blob/main/.github/ISSUE_TEMPLATE/release-check-list.md),
+the plan is to do a succession of component releases, rather than releasing
+all components at the same time. This means that the peer pods release process needs to slot into the correct place.
+The flow of releases should roughly be:
+1. The [guest-components release](https://github.com/confidential-containers/guest-components/releases) (or a pinned
+version is picked) and [trustee release](https://github.com/confidential-containers/trustee/releases) and [kata-containers](https://github.com/kata-containers/kata-containers) bumps these versions in
+    [versions.yaml](https://github.com/kata-containers/kata-containers/blob/main/versions.yaml) under
+    `externals.coco-guest-components.version`, `externals.coco-trustee` and the `image-rs` crate in the agent's
+    [`Cargo.toml`](https://github.com/kata-containers/kata-containers/blob/main/src/agent/Cargo.toml).
```
Comment (mkulke) [inline docs/Release-Process.md:17]: I have a hard time to understand this (the problem could be me not being a native speaker). "[A] ... release ... bumps $something". I have some rough idea what this means, but maybe thsi could be rephrased in simpler terms?

Comment (stevenhorsman) [reply docs/Release-Process.md:17]: Sure, I'll have a go at making it clearer, and your English is fantastic, so it's a me problem, not you!

Comment (stevenhorsman) [reply docs/Release-Process.md]: I hope this helps, otherwise please send it back for more clarification!

**Review by mkulke — APPROVED** [review]

File: docs/Release-Process.md [hunk 4]
```diff
@@ -5,44 +5,67 @@ Containers release
...
+The flow of releases should roughly be:
+1. The [guest-components release](https://github.com/confidential-containers/guest-components/releases) (or a pinned
+version is picked) and [trustee releases](https://github.com/confidential-containers/trustee/releases).
+    - This triggers [kata-containers](https://github.com/kata-containers/kata-containers) to update to these new versions in
+    [versions.yaml](https://github.com/kata-containers/kata-containers/blob/main/versions.yaml) under
+    `externals.coco-guest-components.version`, `externals.coco-trustee` and the `image-rs` crate in the agent's
+    [`Cargo.toml`](https://github.com/kata-containers/kata-containers/blob/main/src/agent/Cargo.toml).
+    - At this point it makes sense for us to stay in sync, by updating the guest-components and kbs that we use in peer pods,
+    by changing the `git.guest-components.reference` and `git.kbs.tag` values in [versions.yaml](../src/cloud-api-adaptor/versions.yaml).
+    We should also bump the kata agent to the latest commit
```
Comment (wainersm) [inline docs/Release-Process.md:21]: Hi @stevenhorsman !

Should we update the agent's version in caa again after kata's release?

Comment (stevenhorsman) [reply docs/Release-Process.md:21]: That's mentioned in L25 & 26?

File: docs/Release-Process.md [hunk 5]
```diff
@@ -5,44 +5,67 @@ Containers release
...
+
+As the [CoCo operator](https://github.com/confidential-containers/operator/) doesn't release until after peer pods,
+the [current plan](https://github.com/confidential-containers/confidential-containers/pull/201#discussion_r1570606331),
+is to pick the latest operator commit to pin that in our released version's instructions of deploying the operator.
+To do this, we should edit the [Makefile](../src/cloud-api-adaptor/Makefile) to replace the
+*github.com/confidential-containers/operator/config/default* and
+*github.com/confidential-containers/operator/config/samples/ccruntime/peer-pods* URLs:
+```
+operator_commit=<latest operator commit sha>
+sed -i "s#\(github.com/confidential-containers/operator/config/default\)#\1?ref=${operator_commit}#" Makefile
```
Comment (wainersm) [inline docs/Release-Process.md:54]: It will end up running:

```
kubectl apply -k "github.com/confidential-containers/operator/config/default?ref=${operator_commit}"
```

Does it work if ${operator_commit}  is a commit SHA1? Asking because I never tried that...
```

Comment (stevenhorsman) [reply docs/Release-Process.md:54]: That's a great point - I'm not sure I've ever tried that either. Something to add to the list. For this alpha release we can pick the v0.9.0-alpha1 tag anyway as the operator has already released

File: docs/Release-Process.md (hunk 4 above)
Comment (wainersm) [reply docs/Release-Process.md:21]: True; never mind.

File: docs/Release-Process.md [hunk 6]
```diff
@@ -5,44 +5,67 @@ Containers release
...
+    - Pre-release testing
+    - Cutting the release
+    - Post release tasks
+1. The CoCo operator updates to use references to the other component releases and then releases itself
+
+### Pre-release testing
+
+In the pre-release/release candidate testing phase
+
+During the pre-release/release candidate phase we we should verify that the kata-containers, guest-components
```
Comment (wainersm) [inline docs/Release-Process.md:43]: s/we we/we

Comment (stevenhorsman) [reply docs/Release-Process.md]: Done. Thanks!

File: docs/Release-Process.md (hunk 5 above)
Comment (wainersm) [reply docs/Release-Process.md:54]: Hi @stevenhorsman !

Nice. Tested it and works with any commit SHA:
```
$ kubectl apply -k "github.com/confidential-containers/operator/config/default?ref=34ebaa6c564f30e7fe94aba8a332a8abe12da020"
namespace/confidential-containers-system created
customresourcedefinition.apiextensions.k8s.io/ccruntimes.confidentialcontainers.org created
serviceaccount/cc-operator-controller-manager created
role.rbac.authorization.k8s.io/cc-operator-leader-election-role created
clusterrole.rbac.authorization.k8s.io/cc-operator-manager-role created
clusterrole.rbac.authorization.k8s.io/cc-operator-metrics-reader created
clusterrole.rbac.authorization.k8s.io/cc-operator-proxy-role created
rolebinding.rbac.authorization.k8s.io/cc-operator-leader-election-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/cc-operator-manager-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/cc-operator-proxy-rolebinding created
configmap/cc-operator-manager-config created
service/cc-operator-controller-manager-metrics-service created
deployment.apps/cc-operator-controller-manager created
```

Comment (stevenhorsman) [reply docs/Release-Process.md:54]: Thank you. You are a ⭐!

File: docs/Release-Process.md [hunk 7]
```diff
@@ -5,44 +5,67 @@ Containers release
...
+To do this, we should edit the [Makefile](../src/cloud-api-adaptor/Makefile) to replace the
+*github.com/confidential-containers/operator/config/default* and
+*github.com/confidential-containers/operator/config/samples/ccruntime/peer-pods* URLs:
+```
+operator_commit=<latest operator commit sha>
+sed -i "s#\(github.com/confidential-containers/operator/config/default\)#\1?ref=${operator_commit}#" Makefile
+sed -i "s#\(github.com/confidential-containers/operator/config/samples/ccruntime/peer-pods\)#\1?ref=${operator_commit}#" Makefile
+```
+
+<!-- TODO, should we worry about updating the e2e test reference in ../src/cloud-api-adaptor/test/provisioner/provision.go too?
```
Comment (wainersm) [inline docs/Release-Process.md:58]: In theory yes. But I'm working on a PR to read from versions.yaml on both Makefile and provision.go. Let's keep this as is.

**Review by wainersm — APPROVED** [review]
Sorry for long time to review it @stevenhorsman . It looks great!

File: docs/Release-Process.md (hunk 7 above)
Comment (stevenhorsman) [reply docs/Release-Process.md:58]: Sounds great. Thanks!
